{
  "cve": {
    "data_type": "CVE",
    "data_format": "MITRE",
    "data_version": "4.0",
    "CVE_data_meta": {
      "ID": "CVE-2019-14232",
      "ASSIGNER": "cve@mitre.org"
    },
    "problemtype": {
      "problemtype_data": [
        {
          "description": [
            {
              "lang": "en",
              "value": "CWE-400"
            }
          ]
        }
      ]
    },
    "references": {
      "reference_data": [
        {
          "url": "https://www.djangoproject.com/weblog/2019/aug/01/security-releases/",
          "name": "https://www.djangoproject.com/weblog/2019/aug/01/security-releases/",
          "refsource": "CONFIRM",
          "tags": [
            "Vendor Advisory"
          ]
        },
        {
          "url": "https://docs.djangoproject.com/en/dev/releases/security/",
          "name": "https://docs.djangoproject.com/en/dev/releases/security/",
          "refsource": "MISC",
          "tags": [
            "Patch",
            "Vendor Advisory"
          ]
        },
        {
          "url": "http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00006.html",
          "name": "openSUSE-SU-2019:1839",
          "refsource": "SUSE",
          "tags": [
            "Third Party Advisory"
          ]
        },
        {
          "url": "https://seclists.org/bugtraq/2019/Aug/15",
          "name": "20190812 [SECURITY] [DSA 4498-1] python-django security update",
          "refsource": "BUGTRAQ",
          "tags": []
        },
        {
          "url": "https://www.debian.org/security/2019/dsa-4498",
          "name": "DSA-4498",
          "refsource": "DEBIAN",
          "tags": []
        },
        {
          "url": "http://lists.opensuse.org/opensuse-security-announce/2019-08/msg00025.html",
          "name": "openSUSE-SU-2019:1872",
          "refsource": "SUSE",
          "tags": []
        },
        {
          "url": "https://security.netapp.com/advisory/ntap-20190828-0002/",
          "name": "https://security.netapp.com/advisory/ntap-20190828-0002/",
          "refsource": "CONFIRM",
          "tags": []
        },
        {
          "url": "https://security.gentoo.org/glsa/202004-17",
          "name": "GLSA-202004-17",
          "refsource": "GENTOO",
          "tags": []
        },
        {
          "url": "http://www.openwall.com/lists/oss-security/2023/10/04/6",
          "name": "[oss-security] 20231004 Django: CVE-2023-43665: Denial-of-service possibility in django.utils.text.Truncator",
          "refsource": "MLIST",
          "tags": []
        },
        {
          "url": "https://groups.google.com/forum/#%21topic/django-announce/jIoju2-KLDs",
          "name": "https://groups.google.com/forum/#%21topic/django-announce/jIoju2-KLDs",
          "refsource": "",
          "tags": []
        },
        {
          "url": "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/STVX7X7IDWAH5SKE6MBMY3TEI6ZODBTK/",
          "name": "FEDORA-2019-647f74ce51",
          "refsource": "",
          "tags": []
        }
      ]
    },
    "description": {
      "description_data": [
        {
          "lang": "en",
          "value": "An issue was discovered in Django 1.11.x before 1.11.23, 2.1.x before 2.1.11, and 2.2.x before 2.2.4. If django.utils.text.Truncator's chars() and words() methods were passed the html=True argument, they were extremely slow to evaluate certain inputs due to a catastrophic backtracking vulnerability in a regular expression. The chars() and words() methods are used to implement the truncatechars_html and truncatewords_html template filters, which were thus vulnerable."
        }
      ]
    }
  },
  "configurations": {
    "nodes": [
      {
        "operator": "OR",
        "cpe_match": [
          {
            "vulnerable": true,
            "cpe23Uri": "cpe:2.3:a:djangoproject:django:*:*:*:*:*:*:*:*"
          },
          {
            "vulnerable": true,
            "cpe23Uri": "cpe:2.3:a:djangoproject:django:*:*:*:*:*:*:*:*"
          },
          {
            "vulnerable": true,
            "cpe23Uri": "cpe:2.3:a:djangoproject:django:*:*:*:*:*:*:*:*"
          }
        ]
      },
      {
        "operator": "OR",
        "cpe_match": [
          {
            "vulnerable": true,
            "cpe23Uri": "cpe:2.3:o:opensuse:leap:15.1:*:*:*:*:*:*:*"
          }
        ]
      }
    ]
  },
  "impact": {
    "baseMetricV2": {
      "cvssV2": {
        "vectorString": "AV:N/AC:L/Au:N/C:N/I:N/A:P",
        "accessVector": "NETWORK",
        "accessComplexity": "LOW",
        "authentication": "NONE",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "PARTIAL",
        "baseScore": 5
      },
      "severity": "MEDIUM",
      "exploitabilityScore": 10,
      "impactScore": 2.9,
      "obtainAllPrivilege": false,
      "obtainUserPrivilege": false,
      "obtainOtherPrivilege": false,
      "userInteractionRequired": false
    }
  },
  "publishedDate": "2019-08-02T15:15Z",
  "lastModifiedDate": "2023-11-07T03:04Z"
}